frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1128,6 +1140,17 @@ func dispatchRun(d *dispatchState, c *instructions.RunCommand, proxy *llb.ProxyE
 }
 
 func dispatchWorkdir(d *dispatchState, c *instructions.WorkdirCommand, commit bool, opt *dispatchOpt) error {
+	if commit {
+		// Only check for workdir lint with committed operations. Does not include
+		// checking the base image.
+		if system.IsAbs(c.Path, d.platform.OS) {
+			d.absWorkdirSet = true
+		} else if !d.absWorkdirSet {
```

Comment:
I think showing this only on first relative `WORKDIR` would be fine as that is the only line that would need to be fixed. So that means `d.absWorkdirSet = true` in all cases.

Reply:
Considering this comment, I modified the code a bit to make it a bit more clear. It still uses `commit` to determine whether to run the lint, but I've changed the variable to `workdirSet` instead and had it unconditionally set to true. I think this is ready to be reviewed and merged if approved now.